### PR TITLE
Pass rule field names in doc level queries during monitor/creation. Remove blocking actionGet() calls 

### PR DESCRIPTION
**Pull request by eirsep**
### Description
Pass rule field names in doc level queries during monitor/creation. This utilizes optimization done in alerting to avoid fetching entire _source of docs in doc level monitor
 
### Issues Resolved
https://github.com/opensearch-project/alerting/issues/1367
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -158,7 +158,7 @@ dependencies {
     implementation group: 'org.apache.commons', name: 'commons-lang3', version: "${versions.commonslang}"
     implementation "org.antlr:antlr4-runtime:4.10.1"
     implementation "com.cronutils:cron-utils:9.1.6"
-    api "org.opensearch:common-utils:${common_utils_version}@jar"
+    api files("/Users/snistala/Documents/opensearch/common-utils/build/libs/common-utils-3.0.0.0-SNAPSHOT.jar")
```
Comment (engechas) [inline build.gradle:161]: Accidental change

File: src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java [hunk 2]
```diff
@@ -92,6 +94,7 @@ public void onResponse(GetMappingsResponse getMappingsResponse) {
                 applyAliasMappings(getMappingsResponse.getMappings(), logType, aliasMappings, partial, new ActionListener<>() {
                     @Override
                     public void onResponse(Collection<CreateMappingResult> createMappingResponse) {
+                        log.error("PERF_DEBUG_SAP: Completed create mappings for {}", indexName);
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java:97]: We should change this to debug and remove the `PERF_DEBUG_SAP` prefix

File: src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java [hunk 3]
```diff
@@ -110,6 +113,7 @@ public void onResponse(Collection<CreateMappingResult> createMappingResponse) {
 
                     @Override
                     public void onFailure(Exception e) {
+                        log.error("PERF_DEBUG_SAP: Failed to create mappings for {}", indexName );
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/mapper/MapperService.java:116]: Same as above, though this one could stay as error

File: src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java [hunk 4]
```diff
@@ -220,19 +222,22 @@ private void checkIndicesAndExecute(
         ActionListener<IndexDetectorResponse> listener,
         User user
     ) {
+        log.error("PERF_DEBUG_SAP: check indices and execute began");
```
Comment (engechas) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportIndexDetectorAction.java:225]: Same as above and throughout this file

**Review by engechas — COMMENTED** [review]
Small comments, rest LGTM

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java [hunk 5]
```diff
@@ -2056,7 +2056,7 @@ public void testCreateDetectorWithCloudtrailAggrRuleWithEcsFields() throws IOExc
         // both req params and req body are supported
         createMappingRequest.setJsonEntity(
                 "{\n" +
-                        "  \"index_name\": \"" + index + "\",\n" +
+                        "  \"index_name\": \"cloudtrail\",\n" +
```
Comment (engechas) [inline src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:2059]: This should still be a variable, right?

File: src/test/java/org/opensearch/securityanalytics/alerts/AlertingServiceTests.java [hunk 6]
```diff
@@ -1,322 +0,0 @@
-/*
```
Comment (engechas) [inline src/test/java/org/opensearch/securityanalytics/alerts/AlertingServiceTests.java:1]: What's the reason to delete this file?

**Review by riysaxen-amzn — DISMISSED** [review]
could we rename the PR description to `Pass rule field names in doc level queries during monitor update/creation.` ?

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java (hunk 5 above)
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/DetectorMonitorRestApiIT.java:2059]: its a test

**Review by engechas — DISMISSED** [review]

**Comment by eirsep** [comment]
> could we rename the PR description to `Pass rule field names in doc level queries during monitor update/creation.` ?

The blocking call removals is also a key fix as that has caused create/update API to get stuck without response. I would prefer retaining it. Open to discuss further if you feel otherwise

**Review by engechas — APPROVED** [review]

File: src/test/java/org/opensearch/securityanalytics/alerts/AlertingServiceTests.java (hunk 6 above)
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/alerts/AlertingServiceTests.java]: twas unintentional.
reverted.
 thanks for flagging this.

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**